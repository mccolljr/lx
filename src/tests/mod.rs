use std::process::Command;

#[test]
fn test_snapshots() {
    glob!("testdata/*.lx", |path| {
        println!("testing {:?}", path);
        let output = Command::new("cargo")
            .arg("run")
            .arg(path)
            .output()
            .expect("failed to execute test");
        let output_str =
            String::from_utf8(output.stdout).expect("oops, output not utf8");
        assert_snapshot!(output_str);
    });
}
