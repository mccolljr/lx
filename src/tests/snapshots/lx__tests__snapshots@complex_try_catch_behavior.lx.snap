---
source: src/tests/mod.rs
expression: output_str
input_file: src/tests/testdata/complex_try_catch_behavior.lx
---
i: 0
i: 1
i: 2
i: 3
i: 4
caught: runtime error: err: i: 5
finally: j: 1
i: 0
i: 1
i: 2
i: 3
i: 4
caught: runtime error: err: i: 5
finally: j: 2
caught outer: runtime error: err: j: 3
do_stuff(): WILL see (finally)
OK!

